src/kernel/emulation/linux/misc/sysctl_machdep.c

```diff
@@ -51,3 +77,132 @@ sysctl_handler(handle_brand_string)
 	return 0;
 }
 
+sysctl_handler(handle_max_basic)
+{
+	setup(0);
+
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+
+	sprintf(old,"%d",eax);
+
+	return 0;
+}
+
+sysctl_handler(handle_family)
+{
+    setup(1);
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+    eax = eax >> 7;
+    eax &= 15;
+
+    sprintf(old,"%d",eax);
+
+    return 0;
+}
+
+sysctl_handler(handle_model)
+{
+    setup(1);
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+    eax = eax >> 3;
+    eax &= 15;
+
+    sprintf(old,"%d",eax);
+
+    return 0;
+}
+
+sysctl_handler(handle_stepping)
+{
+    setup(1);
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+    eax &= 15;
+
+    sprintf(old,"%d",eax);
+
+    return 0;
+}
+
+sysctl_handler(handle_brand_string)
+{
+    unsigned int level = 0;
+    unsigned int eax = 0x80000000;
+
+    unsigned int test[3];
+
+    __cpuid(level,eax,test, test+1,test+2);
+
+    if(eax < 0x80000004) // the information is not implemented
+        return 2;
+
+    eax = 0x80000001;
+
+    union
+    {
+        unsigned int brand[12];
+        char name[49];
+    } v;
+
+    __cpuid(level, eax, v.brand+0x1, v.brand+0x2, v.brand+0x3);
```

Comment:
Like I mentioned in my review above, all the `__cpuid` calls in this function error with "invalid lvalue in asm output". Taking into account my suggestion above, if you need to add the return values to the union, you can do:
```suggestion
    __cpuid(level, eax, ebx, ecx, edx);
    v.brand[0] = eax;
    v.brand[1] = ebx;
    v.brand[2] = ecx;
    v.brand[3] = edx;
```
Same goes for the other two uses of `__cpuid` below.